### PR TITLE
Add eslint rule to enforce PascalCase for enums

### PR DESCRIPTION
**Pull request by thomasdax98**
Add new eslint rule `pascal-case-enums` that enforces using PascalCase for enums

---

Closes COM-378

### CODE REVIEW

File: packages/eslint-plugin/src/rules/pascal-case-enums.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        fixable: "code",
+        messages: {
+            pascalCaseEnumDeclarationMessage: "Enum name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberMessage: "Enum member name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberLiteralMessage: "Enum member literal should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+        },
+    },
+    create(context) {
+        return {
+            "TSEnumDeclaration > Identifier": (identifier) => {
```
Comment (thomasdax98) [inline packages/eslint-plugin/src/rules/pascal-case-enums.ts:19]: Sadly, I couldn't find a way to type `identifier`. It seems like only https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec exports the types. But it's an internal package that isn't publicly published.

File: packages/eslint-plugin/src/rules/pascal-case-enums.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                            name,
+                            pascalCaseName,
+                        },
+                        fix: function (fixer) {
+                            return fixer.replaceText(identifier, pascalCaseName);
+                        },
+                    });
+                }
+            },
+            "TSEnumMember > Literal": (literal) => {
```
Comment (thomasdax98) [inline packages/eslint-plugin/src/rules/pascal-case-enums.ts:55]: Same for `literal` https://github.com/vivid-planet/comet/pull/1631/files#r1469279812

**Comment by johnnyomair** [comment]
As discussed, please verify that this isn't possible with https://typescript-eslint.io/rules/naming-convention/.

**Comment by thomasdax98** [comment]
@johnnyomair I did that in https://github.com/vivid-planet/comet/pull/1656 but I prefer this solution. The main downside is that it's not autofixable.

**Comment by nsams** [comment]
I'm not sure if this should be auto-fixed, as it changes the code incompatible. Usually auto-fixable rules don't change the behavior or API of linted code

**Pull request closed without merging**